docs/source/architecture/journal.rst

```diff
@@ -2,476 +2,196 @@
-   using its own rules.
-
-#. **State Hash Check** - The StateRootHash generated by validating the block is
-   checked against the StateRootHash (state_root_hash field in the BlockHeader)
-   on the block. They must match for the block to be valid.
-
-If the block is computed to be valid, then StateRootHash is committed to the
-store.
 
 The BlockPublisher
 ==================
 
 The BlockPublisher is responsible for creating candidate blocks to extend the
-current chain. The BlockPublisher does all of the housekeeping work around
-creating a block but takes direction from the consensus algorithm for when to
-create a block and when to publish a block.
+current chain. The BlockPublisher does all of the housekeeping work for creating
+a block, but takes direction from the consensus algorithm for when to create a
+block and when to publish a block.
 
-The BlockPublisher follows this logic flow:
+The BlockPublisher responds to the following events:
 
-.. image:: ../images/journal_block_publisher_flow.*
-   :width: 80%
-   :align: center
-   :alt: Journal Block Publisher Diagram
-
-At each processing stage, the consensus algorithm has a chance to inspect and
-confirm the validity of the block.
+- Start block
+- Receive batch
+- Summarize block (stop and make the block available)
+- Finalize block (publish the block)
 
-During CreateBlock, an instance of Consensus.BlockPublisher is created that is
-responsible for guiding the creation of this candidate block.  Also, a
-TransactionScheduler is created and all of the pending Batches are submitted to
-it.
 
-A delay is employed in the checking loop to ensure that there is time for the
-batch processing to occur.
+The ChainController
+===================
 
-Genesis Operation
-=================
+The ChainController is responsible for maintaining and extending the
+blockchain for the validator. This responsibility involves validating proposed
+blocks, evaluating valid blocks to determine if they are the correct chain
+head, and generating new blocks to extend the chain.
 
-The Journal supports Genesis operation. This is the action of creating a root of
-the chain (the Genesis block) when the block store is empty. This operation is
-necessary for bootstrapping a validator network with the desired consensus
-model, any deployment-specific configuration settings, as well as any
-genesis-time transactions for an application's Transaction Family.
+The ChainController determines which chain the validator is currently on and
+coordinates any change-of-chain activities that need to happen.
 
-Genesis Batch Creation
-----------------------
+The ChainController is designed to be able to handle multiple block validation
+activities simultaneously. For instance, if multiple forks form on the network,
+the ChainController can process blocks from all of the competing forks
+simultaneously. This allows progress to be made even when several deep forks are
```

Comment:
This is false. The ChainController is incapable of validating blocks concurrently as it is written today. It can only validate one block at a time.